runtime/src/common/key_format.rs

```diff
@@ -70,9 +68,7 @@ pub trait KeyFormat {
         if data[0] != Self::prefix() {
```

Comment:
come to think of it, we access `[0]` before a length check? that could be risky

Reply:
Nice find, added a check (and also an assert a bit below)

Reply:
thanks!